test/metabase/test/gen2.clj

```diff
@@ -0,0 +1,294 @@
+(ns metabase.test.gen2
+  (:require [clojure.spec.alpha :as s]
+            [clojure.test.check.generators :as gen]
+            [clojure.tools.reader.edn :as edn]
+            [java-time :as t]
+            [metabase.models
+             :refer
+             [Activity Card Collection Dashboard DashboardCard DashboardCardSeries Database Dimension Field
+              FieldValues Metric MetricImportantField NativeQuerySnippet PermissionsGroup PermissionsGroupMembership
+              Pulse PulseCard PulseChannel Table User]]
+            [metabase.test :as mt]
+            [metabase.util.date-2.parse :as u.date]
+            [reifyhealth.specmonstah.core :as rs]
+            [reifyhealth.specmonstah.spec-gen :as rsg]
+            [talltale.core :as tt]
+            [toucan.db :as tdb]))
+
+(def ^:private edn-definitions-dir "./test/metabase/test/data/dataset_definitions/")
+
+(defonce ^:private sample-dataset
+  (edn/read-string {:eof nil, :readers {'t #'u.date/parse}}
+                   (slurp (str edn-definitions-dir "sample-dataset.edn"))))
+
+(defn- random-desc
+  "Return a random name from sample-dataset"
+  []
+  (second (rand-nth (nth (first sample-dataset) 2))))
+
+(defn- coin-toss
+  ([] (coin-toss 0.5))
+  ([p] (< (rand) p)))
+
+;; * items
+(def id-seq (atom 0))
+(s/def ::id (s/with-gen pos-int? #(gen/fmap (fn [_] (swap! id-seq inc)) (gen/return nil))))
+(s/def ::engine #{:postgres :h2})
+(s/def ::not-empty-string (s/and string? not-empty #(< (count %) 10)))
+
+(s/def ::database (s/keys :req-un [::id ::engine ::name ::details]))
+(s/def ::color #{"#A00000" "#FFFFFF"})
+(s/def ::password ::not-empty-string)
+(s/def ::str? (s/or :nil nil? :string string?))
+(s/def ::topic ::not-empty-string)
+(s/def ::details #{ "{}"})
+(s/def ::timestamp #{(t/instant)})
+
+(s/def ::user_id ::id)
+(s/def ::group_id ::id)
+
+(s/def ::email
+  (s/with-gen string?
+    #(gen/fmap (fn [_] (talltale.core/email))
+               (gen/return nil))))
+
+;; (s/def ::description
+;;   (s/with-gen string?
+;;     #(gen/fmap
+;;       (fn [_] (random-desc))
+;;       (gen/return nil))))
+
+(s/def ::first_name
+  (s/with-gen string?
+    #(gen/fmap
+      (fn [_] (talltale.core/first-name))
```

Comment:
You've already aliased this namespace in the `:require` form, any reason not to use that alias here? 

If you don't use the alias it's hard to tell at a glance that you've actually required the namespace, which means the code will work if someone else already required it, but not if you load the namespace first. See https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#require-namespaces-you-use